### DISCUSSION TITLE
Dynamically added input field does not have datepicker

### DISCUSSION THREAD

**Discussion by philharmonie** [livewire/livewire#5541] [1. Help]
I have a `count` attribute in livewire that gets changed dynamically. Based on that, I add date input fields:
```php
<?php for($i=0; $i <= $count; $i++) : ?>
    <input type="text" name="dates[]"  datepicker />
<?php endfor; ?>
```
In `app.js` I have
```javascript
const initiateDatePickers = () => {
 document.querySelectorAll("[datepicker]").forEach(function (datepickerEl) {
   const d = new Datepicker(datepickerEl, datepickerOptions);
  });
}
window.initiateDatePickers = initiateDatePickers;
initiateDatePickers();
```

When I now increase the `count` value, the inputs get added, but they don't have the datepicker.

I tried the following in the view:
```javascript
Livewire.on('reinitiateDatepicker', () => {
   initiateDatePickers();
});
```
and I fire the `$this->emit('reinitiateDatepicker');` in the component when count updates. The `initiateDatePickers()` gets called, but the datepicker are not working.
Generally they are working, because if I add the input directly on the first render, they are there.

**Comment by joshhanley** [comment]
@philharmonie code all looks good to me and if you're `initiateDatePickers()` is actually being called when you fire the event, then it might actually be an issue with the datepicker library.

I know some javascript libraries have a flag like "initialised" once they have been initialised the first time on the page which stops the function running again. But you'd think, as you are passing a new element, that it should work.

Personally though, I would be using Alpine wrap the datepicker and initialise it using Alpine, as it is handled better by Livewire.

Hope this helps!

**Reply by philharmonie** [reply]
@philharmonie  Thank you for taking the time to review my issue. How would that alpine approach look like?
How can I find out, if this is really an issue with the library ([Flowbite datepicker](https://github.com/themesberg/flowbite/blob/main/content/plugins/datepicker.md))

**Comment by joshhanley** [comment]
@philharmonie there is a really basic example in the docs of how to use Alpine with Pikaday https://laravel-livewire.com/docs/2.x/alpine-js#creating-a-datepicker

Your library may require more work than that, but when count changes and a new Alpine component is added to the page, Alpine's `x-init` will fire for that component allowing you to setup the new date picker.

You may also need to make use `wire:ignore` depending on how the library is structure if Livewire keeps removing any DOM elements the library adds. https://laravel-livewire.com/docs/2.x/alpine-js#ignoring-dom-changes

Hope this helps!
